Sources/Brave/States/AppState.swift

```diff
@@ -0,0 +1,232 @@
+public class AppState {
+  public static let shared = AppState()
+
+  public let braveCore: BraveCoreMain
+  public let dau: DAU
+  public let migration: Migration
+  public let profile: Profile
+  public let rewards: BraveRewards
+  
+  public var state: State = .launching(options: [:], active: false) {
+    didSet {
+      switch state {
+      case .launching(_, let isActive):
+        if isActive {
+          // We have to wait until pre 1.12 migration is done until we proceed with database
+          // initialization. This is because Database container may change. See bugs #3416, #3377.
+          DataController.shared.initializeOnce()
+          Migration.postCoreDataInitMigrations()
+          Migration.migrateTabStateToWebkitState(diskImageStore: diskImageStore)
+        }
+        break
+      case .active:
+        break
+      case .backgrounded:
+        break
+      case .terminating:
+        break
+      }
+    }
+  }
+  
+  private(set) public lazy var diskImageStore = { () -> DiskImageStore? in
+    do {
+      return try DiskImageStore(
+        files: profile.files,
+        namespace: "TabManagerScreenshots",
+        quality: UIConstants.screenshotQuality)
+    } catch {
+      Logger.module.error("Failed to create an image store for files: \(self.profile.files.rootPath) and namespace: \"TabManagerScreenshots\": \(error.localizedDescription)")
+    }
+    return nil
+  }()
+
+  private init() {
+    // Setup Constants
+    AppState.setupConstants()
+
+    // Setup BraveCore
+    braveCore = AppState.setupBraveCore().then {
+      $0.scheduleLowPriorityStartupTasks()
+    }
+
+    // Setup DAU
+    dau = DAU(braveCoreStats: braveCore.braveStats)
+    
+    // Setup Profile
+    profile = BrowserProfile(localName: "profile")
+
+    // Setup Migrations
+    migration = Migration(braveCore: braveCore)
+    
+    // Perform Migrations
+    migration.launchMigrations(keyPrefix: profile.prefs.getBranchPrefix(), profile: profile)
+    
+    // Setup Rewards & Ads
+    let configuration = BraveRewards.Configuration.current()
+    Self.migrateAdsConfirmations(for: configuration)
+    rewards = BraveRewards(configuration: configuration)
+    
+    // Setup Custom URL scheme handlers
+    setupCustomSchemeHandlers(profile: profile)
+  }
+
+  public enum State {
+    case launching(options: [UIApplication.LaunchOptionsKey: Any], active: Bool)
+    case active
+    case backgrounded
+    case terminating
+  }
+
+  private static func setupConstants() {
+    // Application Constants must be initialized first
+//    #if MOZ_CHANNEL_RELEASE
+//    AppConstants.buildChannel = .release
+//    #elseif MOZ_CHANNEL_BETA
+//    AppConstants.buildChannel = .beta
+//    #elseif MOZ_CHANNEL_DEV
+//    AppConstants.buildChannel = .dev
+//    #elseif MOZ_CHANNEL_ENTERPRISE
+//    AppConstants.buildChannel = .enterprise
+//    #elseif MOZ_CHANNEL_DEBUG
+//    AppConstants.buildChannel = .debug
+//    #endif
```

Comment:
what about this comment block?

Reply:
I will remove this. It's done in app-delegate.